force-app/main/default/classes/Homework.cls

```diff
@@ -102,6 +129,10 @@ public with sharing class Homework
         // e.g. if value is 10 and power is 2, result should be 100
         // do not use a loop!
 
+        // I feel like there should be a more elegant way of doing this!
+        Double dblValue = (Double) value;
+        result = Math.pow(dblValue, power).intValue(); 
+
```

Comment:
You can take advantage of automatic type promotion here - the compiler will accept an Integer where a Double is expected as a Double can store any Integer value without any loss of accuracy. It doesn't work the other way, as converting a Double to an Integer potentially loses the fractional element, and the value may be too large for an Integer, so you still need to convert the result:

`result=Math.pow(value, power).intValue();`